src/java.base/share/classes/jdk/internal/util/ReferencedKeyMap.java

```diff
@@ -109,7 +108,7 @@ final class ReferencedKeyMap<K, V> implements Map<K, V> {
     private ReferencedKeyMap(boolean isSoft, Map<ReferenceKey<K>, V> map) {
         this.isSoft = isSoft;
         this.map = map;
-        this.stale = new ReferenceQueue<>();
+        this.stale = SharedSecrets.getJavaLangRefAccess().newNativeReferenceQueue();
```

Comment:
I don't think you want all `ReferencedKeyMap` instances use native reference queue which may cause the virtual thread to pin on the carrier thread if it synchronizes the monitor lock.

Perhaps add a new factory method that takes a `ReferenceQueue` and `MethodType` will create it with `NativeReferenceQueue`.